File: src-frontend/src/types/dataset.ts
export interface ColumnStats {
  column: string;
  mean: number;
  std: number;
  min: number;
  max: number;
}

export interface DatasetMeta {
  id: string;
  name: string;
  filename: string;
  plant_part: string;
  target: string;
  n_samples: number;
  n_features: number;
  feature_names: string[];
  columns: string[];
  is_preset: boolean;
}

export interface DatasetDetail extends DatasetMeta {
  preview: Record<string, number>[];
}

export interface DatasetStatsResponse {
  dataset_id: string;
  stats: ColumnStats[];
}

File: src-backend/app/models/dataset.py
from pydantic import BaseModel


class ColumnStats(BaseModel):
    column: str
    mean: float
    std: float
    min: float
    max: float


class DatasetMeta(BaseModel):
    id: str
    name: str
    filename: str
    plant_part: str
    target: str
    n_samples: int
    n_features: int
    feature_names: list[str]
    columns: list[str]
    is_preset: bool


class DatasetDetail(DatasetMeta):
    preview: list[dict[str, float]]


class DatasetStatsResponse(BaseModel):
    dataset_id: str
    stats: list[ColumnStats]


class UpdateTargetRequest(BaseModel):
    target_column: str
